stm32Transceiver/button_demo_trx/Core/Src/main.c

```diff
@@ -0,0 +1,356 @@
+  */
+/* USER CODE END Header */
+/* Includes ------------------------------------------------------------------*/
+#include "main.h"
+
+/* Private includes ----------------------------------------------------------*/
+/* USER CODE BEGIN Includes */
+#include "string.h"
+#include "NRF24L01.h"
+/* USER CODE END Includes */
+
+/* Private typedef -----------------------------------------------------------*/
+/* USER CODE BEGIN PTD */
+
+/* USER CODE END PTD */
+
+/* Private define ------------------------------------------------------------*/
+/* USER CODE BEGIN PD */
+/* USER CODE END PD */
+
+/* Private macro -------------------------------------------------------------*/
+/* USER CODE BEGIN PM */
+
+/* USER CODE END PM */
+
+/* Private variables ---------------------------------------------------------*/
+SPI_HandleTypeDef hspi3;
+
+UART_HandleTypeDef huart2;
+
+/* USER CODE BEGIN PV */
+uint8_t RxAddress[] = {0xEE,0xDD,0xCC,0xBB,0xAA}; // list of data pipes available
+uint8_t RxData[32];
+uint8_t txData[32];
+uint8_t TxAddress[] = {0xEE,0xDD,0xCC,0xBB,0xAA};
+uint8_t uButtonFlag = 0;
+/* USER CODE END PV */
+
+/* Private function prototypes -----------------------------------------------*/
+void SystemClock_Config(void);
+static void MX_GPIO_Init(void);
+static void MX_USART2_UART_Init(void);
+static void MX_SPI3_Init(void);
+/* USER CODE BEGIN PFP */
+uint8_t userButtonActive(void);
+/* USER CODE END PFP */
+
+/* Private user code ---------------------------------------------------------*/
+/* USER CODE BEGIN 0 */
+
+/* USER CODE END 0 */
+
+/**
+  * @brief  The application entry point.
+  * @retval int
+  */
+int main(void)
+{
+  /* USER CODE BEGIN 1 */
+	// set up tx data
+	char txBuf[32];
+	strcpy(txBuf,  "According to all known laws of a");
+
+	// convert to transmittable form:
+	memcpy(&txData, txBuf, 32); // done! data ready in txData
+  /* USER CODE END 1 */
+
+  /* MCU Configuration--------------------------------------------------------*/
+
+  /* Reset of all peripherals, Initializes the Flash interface and the Systick. */
+  HAL_Init();
+
+  /* USER CODE BEGIN Init */
+
+  /* USER CODE END Init */
+
+  /* Configure the system clock */
+  SystemClock_Config();
+
+  /* USER CODE BEGIN SysInit */
+
+  /* USER CODE END SysInit */
+
+  /* Initialize all configured peripherals */
+  MX_GPIO_Init();
+  MX_USART2_UART_Init();
+  MX_SPI3_Init();
+  /* USER CODE BEGIN 2 */
+  NRF24_Init();
+  NRF24_RxMode(RxAddress, 6); // receive on channel 6
+  /* USER CODE END 2 */
+
+  /* Infinite loop */
+  /* USER CODE BEGIN WHILE */
+  while (1)
+  {
+	  // push button - transmit bee movie
+	  if(userButtonActive())
+	  {
+		  // user wants to transmit
+		  // set up tx mode:
+		  NRF24_TxMode(TxAddress, 6); // transmit on channel 6
+		  HAL_Delay(1); // let SPI settle
+		  if(NRF24_Transmit(txData)) // transmit 32B msg, check if it worked
+		  {
+			  // flash the happy light: 3 quick blinks
+			  int flashCount = 3;

```

Comment:
variables should ideally be located in one place, move flashcount to a user private variable or user define section above.